File: web/src/api/device.js
import service from '@/utils/request'

export function getDeviceList(params) {
  return service({
    url: '/device/list',
    method: 'get',
    params
  })
}

export function getDevice(id) {
  return service({
    url: `/device/${id}`,
    method: 'get'
  })
}

export function createDevice(data) {
  return service({
    url: '/device',
    method: 'post',
    data
  })
}

export function updateDevice(id, data) {
  return service({
    url: `/device/${id}`,
    method: 'put',
    data
  })
}

export function deleteDevice(id) {
  return service({
    url: `/device/${id}`,
    method: 'delete'
  })
}

export function getDeviceChannels(id) {
  return service({
    url: `/device/${id}/channels`,
    method: 'get'
  })
}

export function queryDeviceCatalog(id) {
  return service({
    url: `/device/${id}/catalog`,
    method: 'post'
  })
}

export function getDeviceStatus(id) {
  return service({
    url: `/device/${id}/status`,
    method: 'get'
  })
}

File: server/router/gb28181/device.go
package gb28181

import (
	"wvp-go/server/api/v1/gb28181"

	"github.com/gin-gonic/gin"
)

type DeviceRouter struct{}

var deviceRouter = new(DeviceRouter)

func (d *DeviceRouter) InitDeviceRouter(Router *gin.RouterGroup) {
	deviceRouterGroup := Router.Group("/device")
	{
		deviceRouterGroup.GET("/list", gb28181.GetDeviceList)
		deviceRouterGroup.GET("/:id", gb28181.GetDevice)
		deviceRouterGroup.POST("", gb28181.CreateDevice)
		deviceRouterGroup.PUT("/:id", gb28181.UpdateDevice)
		deviceRouterGroup.DELETE("/:id", gb28181.DeleteDevice)
		deviceRouterGroup.GET("/:id/channels", gb28181.GetDeviceChannels)
		deviceRouterGroup.POST("/:id/catalog", gb28181.QueryDeviceCatalog)
		deviceRouterGroup.GET("/:id/status", gb28181.GetDeviceStatus)
	}

	channelRouterGroup := Router.Group("/channel")
	{
		channelRouterGroup.GET("/list", gb28181.GetChannelList)
	}
}
